csv.go

```diff
@@ -0,0 +1,57 @@
+package middlewares
+
+import (
+	"bytes"
+	"encoding/json"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+func csvFromJSON(b []byte) (result []byte, err error) {
+	var data []map[string]interface{}
+	err = json.Unmarshal(b, &data)
+	if err != nil {
+		return
+	}
+	if len(data) == 0 {
+		return
+	}
+	keys := make([]string, 0, len(data[0]))
+	for k := range data[0] {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	var strOut bytes.Buffer
+	line := strings.Join(keys, ";") + "\n"
+	strOut.WriteString(line)
+	for _, v := range data {
+		values := decodeLine(keys, v)
+		line := strings.Join(values, ";") + "\n"
+		strOut.WriteString(line)
+	}
+	return strOut.Bytes(), nil
+}
+
+func decodeLine(keys []string, data map[string]interface{}) []string {
```

Comment:
**encoding/csv** does this job for you
https://golang.org/src/encoding/csv/writer.go#L48